ApiIndex/index.proto

```diff
@@ -0,0 +1,183 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package google.tools.apiindex.v1;
+
+option csharp_namespace = "Google.Cloud.Tools.ApiIndex.V1";
+
+// The overall API index. The index file
+// is a JSON representation of this message.
+// See the glossary for more details about the
+// terminology used in this schema.
+message Index {
+  // All the known APIs, ordered lexicographically by ID.
+  repeated Api apis = 1;
+  
+  // Metadata about the current API index.
+  IndexMetadata metadata = 2;
```

Comment:
What is the relationship between this metadata and the package version `google.tools.apiindex.v1`. Am I right in thinking that they will move at the same pace and IndexMetadata is just there so that it's programatically easy to know the version of the generated index?

Reply:
There's one difference, in terms of compatibility: it's possible that `google.tools.apiindex.v2.IndexMetadata` could have more information, but we *mustn't* make breaking changes within the `IndexMetadata` message. (We could *potentially* deprecate something in v2 and remove it in v3, but I hope we don't need to.)

Tools should be able to read any version of the schema by ignoring JSON fields they don't understand, but they should always be able to understand "enough" of the metadata to find out the version and report if it's deprecated.